File: src/main/java/me/nazarxexe/survival/econ/EconomyRegistrations.java
package me.nazarxexe.survival.econ;

import cn.nukkit.event.EventHandler;
import cn.nukkit.event.Listener;
import cn.nukkit.event.player.PlayerJoinEvent;
import cn.nukkit.event.player.PlayerQuitEvent;
import org.jetbrains.annotations.NotNull;

public class EconomyRegistrations implements Listener {

    final Economy plugin;
    final EconomyAPI api;


    public EconomyRegistrations(@NotNull Economy plugin) {
        this.plugin = plugin;
        this.api = plugin.getApi();
    }

    @EventHandler
    public void onJoin(@NotNull PlayerJoinEvent e){
        api.databaseToCached(e.getPlayer());
    }

    @EventHandler
    public void onLeft(@NotNull PlayerQuitEvent e){
        api.cachedToDatabase(e.getPlayer());
    }

}

File: src/main/java/me/nazarxexe/survival/econ/EconomyRegistrations.kt
package me.nazarxexe.survival.econ

import cn.nukkit.event.EventHandler
import cn.nukkit.event.Listener
import cn.nukkit.event.player.PlayerJoinEvent
import cn.nukkit.event.player.PlayerQuitEvent

class EconomyRegistrations(plugin: Economy) : Listener {
    private val plugin: Economy?
    private val api: EconomyAPI?

    init {
        this.plugin = plugin
        this.api = this.plugin.api
    }

    @EventHandler
    fun onJoin(e: PlayerJoinEvent) {
        this.api?.databaseToCached(e.player);
    }

    @EventHandler
    fun onLeft(e: PlayerQuitEvent) {
        api?.cachedToDatabase(e.player)
        api?.cache?.remove(e.player.uniqueId)
    }
}